### PR TITLE
Feat/ofv merged

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

This PR adds the OFV Market Resolver tool introduced in #225.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by 0xArdi** [comment]
@kongzii @victorpolisetty Can you take a look a this PR?
It's adding the OFV Market Resolver tool without the dependencies.

**Comment by victorpolisetty** [comment]
lgtm! I ran it with my own openai and serper api key and it worked well. Gave the answer based on the latest polls for the election.

Screenshot 2024-08-07 at 5 14 08 PM

**Review by victorpolisetty — APPROVED** [review]

File: packages/gnosis/customs/ofv_market_resolver/ofv_market_resolver.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    `Former Trump Organization CFO Allen Weisselberg was sentenced to jail by 15 April 2024.`
+    """
+    llm = ChatOpenAI(
+        model=model,
+        temperature=0.0,
+        api_key=openai_api_key,
+    )
+
+    prompt = f"""
+Rewrite the question into a simple annoucment sentence stating a fact or prediction like it is already known.
```
Comment (victorpolisetty) [inline packages/gnosis/customs/ofv_market_resolver/ofv_market_resolver.py:126]: nit: type in word "announcement"

**Review by victorpolisetty — APPROVED** [review]

**Comment by kongzii** [comment]
Looks good; dependency changes were already merged in [separate PR](https://github.com/valory-xyz/mech/pull/243), so that's fine. And if you are going to merge this PR as it is now,  then https://github.com/valory-xyz/mech/pull/244 can be closed, as this PR contains these changes.

File: pyproject.toml [hunk 2]
```diff
@@ -76,6 +76,7 @@ langgraph = "==0.0.55"
 langchain-community = "==0.2.1"
 prediction-market-agent-tooling = "^0.39.0"
 google-generativeai = "==0.6.0"
+openfactverification = { git = "https://github.com/kongzii/OpenFactVerification.git", rev = "5f54b916e20503b7cbef99dcb0eceb2738ae5451" }
```
Comment (kongzii) [inline pyproject.toml:79]: Noticed this just now, I released this as package on PyPi as requested, no need to keep it from git: https://pypi.org/project/openfactverification-kongzii/

File: packages/gnosis/customs/ofv_market_resolver/__init__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the bet amount per threshold strategy."""
```
Comment (evangriffiths) [inline packages/gnosis/customs/ofv_market_resolver/__init__.py:20]:
```suggestion
"""This module contains the ofv market resolver."""
```

**Review by dvilelaf — APPROVED** [review]

**Pull request merged**